DCOS-55889 show tooltip if task is not existent

pierrebeitz opened the pull request:
Mesos garbage collects tasks. We don't want to tasks that are not present anymore in Mesos. We want to show a tooltip instead.

## Review

commit-by-commit

## Testing

You most likely want a DCOS with a Job that has both kinds of tasks. It looks like a pretty good one is this job on SOAK: http://localhost:4200/#/jobs/detail/spark-hdfs-kerberos-terasort.

Make sure that we don't link to old tasks and that we link to tasks when they are present. Also make sure there are no weird states when reloading the page.

## Screenshots

<img width="1290" alt="Monosnap 2019-07-30 17-19-34" src="https://user-images.githubusercontent.com/300861/62142274-4de00b80-b2ee-11e9-8937-70eff16b27d2.png">
<img width="1335" alt="OS Enterprise 2019-07-30 17-20-25" src="https://user-images.githubusercontent.com/300861/62142323-6b14da00-b2ee-11e9-818e-1300d80c4a8a.png">

Closes DCOS-55889

GeorgiSTodorov commented on plugins/services/src/js/components/SDKPlans.tsx:
```diff
@@ -127,7 +127,11 @@ const SDKPlans = componentFromStream(props$ => {
     ),
-    catchError(() => of(<ErrorScreen />)),
+    catchError(e => {
+      console.warn(e);
```
Build is failing because of this line.

pierrebeitz replied:
removed the according commit for now to come up with proper logging-facilities.

GeorgiSTodorov commented on plugins/jobs/src/js/pages/JobRunHistoryTable.js:
```diff
@@ -246,17 +256,39 @@ class JobRunHistoryTable extends React.Component {
+          trigger={<span>{taskID}</span>}
+        >
+          <Trans>
+            The data related to this tasks has already been cleaned up.
```
These* tasks.

pierrebeitz replied:
fixed!

GeorgiSTodorov commented on plugins/jobs/src/js/pages/JobTaskDetailPage.js:
```diff
@@ -30,6 +59,16 @@ class JobTaskDetailPage extends React.Component {
+    if (!task) {
+      return (
+        <Trans>
+          Either the data related to that task has already been cleaned up or
```
I think "this" instead of "that" sounds better.

pierrebeitz replied:
totally agree!

GeorgiSTodorov commented on plugins/jobs/src/js/pages/JobTaskDetailPage.js:
```diff
@@ -30,6 +59,16 @@ class JobTaskDetailPage extends React.Component {
+      return (
+        <Trans>
+          Either the data related to that task has already been cleaned up or
+          the given task-id does not exist.
```
I think that "task-id" should be written without a dash.

pierrebeitz replied:
i made it `Task ID` now as it looks like a proper noun to me

GeorgiSTodorov commented on plugins/jobs/src/js/pages/JobRunHistoryTable.js:
```diff
@@ -246,17 +248,34 @@ class JobRunHistoryTable extends React.Component {
+      );
+      const hint = (
+        <Tooltip
+          preferredDirections={["top-left", "bottom-left"]}
```
What do you think about only having "bottom-left" here so that we don't hide the job id?
### Currently
![Screenshot from 2019-07-31 09-53-22](https://user-images.githubusercontent.com/40791275/62190148-529dd180-b379-11e9-9029-afe8b911bc39.png)

### My suggestion
![Screenshot from 2019-07-31 09-54-04](https://user-images.githubusercontent.com/40791275/62190155-57628580-b379-11e9-98a5-dd323bd6741f.png)

pierrebeitz replied:
really good catch!

GeorgiSTodorov commented:
I left some comments. Other than that, it looks good.

GeorgiSTodorov commented on plugins/jobs/src/js/pages/JobRunHistoryTable.js:
```diff
@@ -246,17 +256,39 @@ class JobRunHistoryTable extends React.Component {
+      // It may be that the data only *seems* to be present because we don't
+      // remove tasks from the MesosStateStore when they're garbage collected in
+      // the background.
+      const dataSeemsStillPresent = !!MesosStateStore.getTaskFromTaskID(taskID);
```
What do you think about something like:
```
const content = !!MesosStateStore.getTaskFromTaskID(taskID) ?
  <Link
  ...
  :
  <Tooltip
  ...
```
since we never need both at the same time.

pierrebeitz replied:
i thought the same at some point in time but must have forgotten about it again...
it's changed!

pierrebeitz commented:
@GeorgiSTodorov 	thanks for the awesome review! i think i addressed all of the points. 

regarding `Edit: I ran the extract command and there were changes.`: i see them as well but did not commit them as i thought it might be strange to see those changes popping up in this PR. maybe we should fix this directly on master.

GeorgiSTodorov commented:
@pierrebeitz , thank you for addressing my feedback so well and so fast. I will take a look again when I am done with another task.

> regarding Edit: I ran the extract command and there were changes.: i see them as well but did not commit them as i thought it might be strange to see those changes popping up in this PR. maybe we should fix this directly on master.

We had some issues with the extraction of these strings, and maybe you started working on your PR before these issues were fixed.

GeorgiSTodorov commented:
@pierrebeitz , do you want a second review or should we merge when it is green?

pierrebeitz commented:
i think we can merge once green.

nLight commented:
@izharbuendia please PDUX validate
